system/lib/libcxx/new.cpp

```diff
@@ -74,8 +74,12 @@ operator new(std::size_t size) _THROW_BAD_ALLOC
         else
 #ifndef _LIBCPP_NO_EXCEPTIONS
             throw std::bad_alloc();
+#else
+#ifdef __EMSCRIPTEN__
```

Comment:
Can you add a comment here saying why we modify the code here directly rather than installing a handler?    e.g. because we care about the extra 1% code side it adds to the empty C++ program :)